packages/clerk-js/src/ui/router/PathRouter.tsx

```diff
@@ -34,15 +35,22 @@ export const PathRouter = ({ basePath, preservedParams, children }: PathRouterPr
   };
 
   React.useEffect(() => {
-    const effect = async () => {
+    const convertHashToPath = async () => {
       if (window.location.hash.startsWith('#/')) {
-        const hashToPath = new URL(window.location.pathname + window.location.hash.substr(1), window.location.href);
+        const hashToPath = new URL(window.location.pathname + window.location.hash.substring(1), window.location.href);
         hashToPath.pathname = trimTrailingSlash(hashToPath.pathname);
+
+        //transfer search params
+        const url = new URL(window.location.href);
+        for (const [key, value] of url.searchParams) {
+          hashToPath.searchParams.append(key, value);
```

Comment:
There is similar logic [here](https://github.com/clerkinc/javascript/blob/583188f80ea36b5aa94e252b02c022bb4240aab8/packages/clerk-js/src/utils/url.ts#L150). Should we extract this to a utility function ?

Reply:
@desiprisg 
Shall we try this approach? 
```ts
/**
 * Creates a new URL by merging a fragment-based URL, if found.
 * The result URL has the original and the fragment pathnames appended 
 * and also includes all search params from both places.
 * 
 * @example
 * Input
 * https://accounts.clerk.com/sign-in?user_param=hello#/verify/factor-one?redirect_url=/protected
 * Return value: 
 * https://accounts.clerk.com/sign-in/verify/factor-one?user_param=hello&redirect_url=/protected
 */
const mergeFragmentUrl = (_url: string | URL) => {
    const url = new URL(_url);
    const hasUrlInFragment = url.hash.startsWith('#/')

    if(!hasUrlInFragment) {
        return url
    }

    const fragmentUrl = new URL(url.hash.replace('#/','/'), url.href)
    const mergedPathname = [url.pathname, fragmentUrl.pathname]
        .map(s => s.split('/'))
        .flat()
        .filter(Boolean)
        .join('/')

    const mergedUrl  = new URL(mergedPathname, url)
    url.searchParams.forEach((val, key) => { mergedUrl.searchParams.set(key, val) })
    fragmentUrl.searchParams.forEach((val, key) => { mergedUrl.searchParams.set(key, val) })
    return mergedUrl;
}
```